astroquery/astrometry_net/core.py

```diff
@@ -0,0 +1,618 @@
+    ``allow_modifications``:
+        Whether or not images uploaded to astrometry.net are allowed to be modified by
+        other users. This can either be set to ``y``, ``n``, or ``d``, which 
+        uses the default license associated with the api key.
+"""
+
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+from __future__ import print_function
+
+# put all imports organized as shown below
+# 1. standard library imports
+
+# 2. third party imports
+import astropy.units as u
+import astropy.coordinates as coord
+import astropy.io.votable as votable
+from astropy.table import Table
+from astropy.io import fits
+from astropy import log
+
+# 3. local imports - use relative imports
+# commonly required local imports shown below as example
+from ..query import BaseQuery # all Query classes should inherit from this.
+from ..utils import commons # has common functions required by most modules
+from ..utils import prepend_docstr_noreturns # automatically generate docs for similar functions
+from ..utils import async_to_sync # all class methods must be callable as static as well as instance methods.
+from . import SERVER, TIMEOUT # import configurable items declared in __init__.py
+import time
+
+# export all the public classes and methods
+__all__ = ['AstrometryNet', 'AstrometryNetClass']
+
+# declare global variables and constants if any
+
+# Now begin your main class
+# should be decorated with the async_to_sync imported previously
+
+astrometry_net_url = 'http://nova.astrometry.net/'
+apiurl = 'http://nova.astrometry.net/api/'
+
+@async_to_sync
+class AstrometryNetClass(BaseQuery):
+
+    """
+    Not all the methods below are necessary but these cover most of the common cases, new methods may be added if necessary, follow the guidelines at <http://astroquery.readthedocs.org/en/latest/api.html>
+    """
+    # use the Configuration Items imported from __init__.py to set the URL, TIMEOUT, etc.
+    URL = SERVER()
+    TIMEOUT = TIMEOUT()
+
+    def _store_API_key(self, value):
+        """ Cache the Astrometry.net API key on disk. """
+        pass
+
+    def _get_stored_API_key(self):
+        """ Return the API key, raise KeyError if not cached on disk. """
+        raise KeyError
+
+    @property
+    def key(self):
+        """ Getter for the Astrometry.net API key. """
+
+        if self._key is None:
+            log.error("Astrometry.net API key not found")
+        return self._key
+
+    @key.setter
+    def key(self, value):
+        """ Setter for the API key, cache it on disk. """
+
+        self._store_API_key(value)
+        self._key = value
+
+    def __init__(self):
+        """ Load the cached API key, show a warning message if can't be found. """
+
+        try:
+            self._key = self._get_stored_API_key()
+        except KeyError:
+            log.warn("Astrometry.net API key not found")
+            log.warn("You need to register it with Astrometry.key = 'XXXXXXXX'")
```

Comment:
will astrometry.net work without an API key?  If not, then this should be an Exception, not a warning.